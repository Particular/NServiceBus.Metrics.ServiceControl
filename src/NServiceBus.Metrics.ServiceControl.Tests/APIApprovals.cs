using System.Runtime.CompilerServices;
using NServiceBus;
using NServiceBus.Metrics.ServiceControl.Msmq.Tests;
using NUnit.Framework;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public void Approve()
    {
        var publicApi = ApiGenerator.GeneratePublicApi(typeof(MetricsOptionsExtensions).Assembly);
        TestApprover.Verify(publicApi);
    }
}