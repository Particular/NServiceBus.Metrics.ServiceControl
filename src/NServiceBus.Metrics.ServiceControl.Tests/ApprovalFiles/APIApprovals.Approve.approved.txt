[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.2", FrameworkDisplayName=".NET Framework 4.5.2")]
namespace NServiceBus.Metrics.ServiceControl
{
    public interface IReportNativeQueueLength
    {
        System.Collections.Generic.IEnumerable<string> MonitoredQueues { get; }
        void ReportQueueLength(string physicalQueueName, long queueLength);
    }
}
namespace NServiceBus
{
    public class static MetricsOptionsExtensions
    {
        public static void SendMetricDataToServiceControl(this NServiceBus.MetricsOptions options, string serviceControlMetricsAddress, System.TimeSpan interval, string instanceId = null) { }
        public static void SetServiceControlMetricsMessageTTBR(this NServiceBus.MetricsOptions options, System.TimeSpan timeToBeReceived) { }
    }
}